src/HandleMgr.cpp

```diff
@@ -78,6 +79,16 @@ namespace splash
         this->fileAccProperties = fileAccProperties;
         this->fileFlags = flags;
         this->singleFile = false;
+        // Validation: For parallel files we normally append MPI rank or iteration number.
+        //             This is disabled by using FNS_FULLNAME
+        //             or when the filename already contains an h5-extension.
+        if (fileNameScheme != FNS_FULLNAME && baseFilename.find(".h5") != std::string::npos)
+        {
+            log_msg(1, "\n"
```

Comment:
I wanted to make this a warning, not an error as there might be (whatever) reasons for using a full filename without setting the `fileNameScheme`.
I also removed the additional check for `mpiSize` as you might e.g. have multiple processes write a different iteration each. Then mpiSize is 1 but they still may erroneously access the same file.